src/Shell/RedirectClipboardTrait.php

```diff
@@ -0,0 +1,162 @@
+    }
+
+    /**
+     * Tests if a command line tool can be found on Linux.
+     *
+     * @param string $cmd The command to locate.
+     * @return bool
+     */
+    private static function which($cmd)
+    {
+        $code = 0;
+        $output = null;
+        exec("which {$cmd}", $output, $code);
+        return $code == 0;
+    }
+
+    /**
+     * Sends the temporary file to the clipboard.
+     *
+     * @return bool
+     */
+    private static function sendClipboardFile()
+    {
+        if (self::$_clipboardFile == null) {
+            return false;
+        }
+        $fileName = self::$_clipboardTemp;
+
+        if (strtoupper(substr(PHP_OS, 0, 3)) === 'WIN') {
+            // Windows
+            return trim(shell_exec("clip < {$fileName}")) == '';
+        }
+        if (self::which('pbcopy')) {
+            // OSX
+            return trim(shell_exec("pbcopy < {$fileName}")) == '';
+        }
+        if (self::which('xclip')) {
+            // Linux
+            return trim(shell_exec("xclip -selection clipboard < {$fileName}")) == '';
+        }
+        return false;
+    }
+
+    /**
+     * Closes the temporary file and deletes it.
+     */
+    private static function closeClipboardFile()
+    {
+        if (self::$_clipboardFile == null) {
+            return;
+        }
+        self::$_clipboardFile->close();
+        self::$_clipboardFile = null;
+        if (file_exists(self::$_clipboardTemp)) {
+            unlink(self::$_clipboardTemp);
+        }
+    }
+
+    /**
+     * Runs the Shell with the provided argv.
+     *
+     * @param array $argv Array of arguments.
+     * @param bool $autoMethod Set to true to allow any public methods
+     * @param array $extra Extra parameters.
+     * @return mixed
+     * @see \Cake\Console\Shell::runCommand
+     */
+    public function runCommand($argv, $autoMethod = false, $extra = [])
+    {
+        if (!self::openClipboardFile()) {
+            $this->abort('Could not open temp file required for clipboard.');
+        }
+
+        try {
+            $res = parent::runCommand($argv, $autoMethod = false, $extra = []);
+            if (self::sendClipboardFile()) {
+                $this->_io->out('<success>Copy to clipboard</success>');
+            } else {
+                $this->_io->out('<error>Failed to send to clipboard</error>');
+                $this->_io->out('');
+                $this->_io->out('Linux users need to install xclip.');
+                $this->_io->out('   sudo apt-get install xclip');
+            }
+        } finally {
+            self::closeClipboardFile();
+        }
+
+        return $res;
+    }
+
+    /**
+     * Optionally redirects file output to the clipboard.
+     *
+     * @param string $path Where to put the file.
+     * @param string $contents Content to put in the file.
+     * @return bool Success
+     */
+    public function createFile($path, $contents)
```

Comment:
There's no option right now in Bake to pick which of the set of files it bakes. So if the bake operation generates 3 files. I have to send them all to the clipboard.